zebra-chain/src/orchard/commitment.rs

```diff
@@ -141,10 +141,11 @@ impl NoteCommitment {
 
         let rcm = CommitmentRandomness::from(note.rseed);
 
-        Some(NoteCommitment::from(
-            sinsemilla_commit(rcm.0, b"z.cash:Orchard-NoteCommit", &s)
-                .expect("valid orchard note commitment, not ⊥ "),
-        ))
+        Some(NoteCommitment::from(sinsemilla_commit(
+            rcm.0,
+            b"z.cash:Orchard-NoteCommit",
+            &s,
+        )?))
```

Comment:
[Section 4.1.8](https://zips.z.cash/protocol/protocol.pdf#abstractcommit) of the protocol spec says:
![image](https://user-images.githubusercontent.com/643204/174988124-09985e2e-e9fa-42c9-b945-86e44daa39c1.png)

It appears as though with this change, the case of an invalid diversifier (the `?` on line 127) is being conflated with a $\bot$ or 0 return from $\mathsf{Commit^{ivk}}$. ($\bot$ and 0 are not distinguished from each other in [section 4.2.3](https://zips.z.cash/protocol/protocol.pdf#orchardkeycomponents).)

Reply:
reverted in https://github.com/ZcashFoundation/zebra/pull/4667/commits/984ee1ce82b1144210c697444797555f2fa62a64